terragrunt/modules/sqs/variables.tf

```diff
@@ -0,0 +1,55 @@
+variable "delay_seconds" {
+  description = "The time in seconds that the delivery of all messages in the queue will be delayed."
+  type        = number
+  default     = 0
+}
+
+variable "max_message_size" {
+  description = "The limit of how many bytes a message can contain before Amazon SQS rejects it."
+  type        = number
+  default     = 262144 # 256 KiB
+}
+
+variable "max_receive_count" {
+  default     = 10
+  description = ""
+  type        = number
+}
+
+variable "message_retention_seconds" {
+  description = "The number of seconds Amazon SQS retains a message."
+  type        = number
+  default     = 345600 # 4 Days
```

Comment:
We shouldn't delete until the message is consumed.

Reply:
It's impossible, The max amount of time you can hold on to it is 2 weeks. 
Hence existence of DLQs 
[4 days is AWS default, which seems reasonable in most designs](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_SetQueueAttributes.html).